internal/featuretests/replaceprefix_test.go

```diff
@@ -0,0 +1,577 @@
+						{Prefix: "/v2/container-release", Replacement: "/artifactory/api/docker/container-release/v2"},
+						{Prefix: "/v2/container-external", Replacement: "/artifactory/api/docker/container-external/v2"},
+						{Prefix: "/v2/container-public", Replacement: "/artifactory/api/docker/container-public/v2"},
+					},
+				},
+			}},
+		},
+	})
+
+	rh.OnAdd(&projcontour.HTTPProxy{
+		ObjectMeta: meta("artifactory/artifactory"),
+		Spec: projcontour.HTTPProxySpec{
+			VirtualHost: &projcontour.VirtualHost{
+				Fqdn: "artifactory.projectcontour.io",
+			},
+			Includes: []projcontour.Include{
+				{Name: "routes", Conditions: prefixCondition("/v2/container-sandbox")},
+				{Name: "routes", Conditions: prefixCondition("/v2/container-release")},
+				{Name: "routes", Conditions: prefixCondition("/v2/container-external")},
+				{Name: "routes", Conditions: prefixCondition("/v2/container-public")},
+			},
+		},
+	})
+
+	c.Request(routeType).Equals(&v2.DiscoveryResponse{
+		Resources: resources(t,
+			envoy.RouteConfiguration("ingress_http",
+				envoy.VirtualHost("artifactory.projectcontour.io",
+
+					envoy.Route(routePrefix("/v2/container-sandbox/"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-sandbox/v2/"),
+					),
+					envoy.Route(routePrefix("/v2/container-sandbox"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-sandbox/v2"),
+					),
+					envoy.Route(routePrefix("/v2/container-release/"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-release/v2/"),
+					),
+					envoy.Route(routePrefix("/v2/container-release"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-release/v2"),
+					),
+					envoy.Route(routePrefix("/v2/container-public/"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-public/v2/"),
+					),
+					envoy.Route(routePrefix("/v2/container-public"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-public/v2"),
+					),
+					envoy.Route(routePrefix("/v2/container-external/"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-external/v2/"),
+					),
+					envoy.Route(routePrefix("/v2/container-external"),
+						withPrefixRewrite(routeCluster("artifactory/service/8080/da39a3ee5e"),
+							"/artifactory/api/docker/container-external/v2"),
+					),
+				),
+			),
+			envoy.RouteConfiguration("ingress_https"),
+		),
+		TypeUrl: routeType,
+	})
+}
+
+func TestHTTPProxyPathPrefix(t *testing.T) {
+	subtests := []struct {
+		Name string
+		Func func(*testing.T)
+	}{
+		{Name: "Basic", Func: basic},
```

Comment:
Does this indirection add much value? What about just calling the `basic` function, `TestHTTPProxyPathPrefixBasic`?

Reply:
I think it has a sliver of value. It groups the tests better and makes it easier to run the group. I don't think there's much downside.